src/web/app/components/user-notifications-list/user-notifications-list.component.spec.ts

```diff
@@ -26,11 +63,126 @@ describe('UserNotificationsListComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(UserNotificationsListComponent);
+    notificationService = TestBed.inject(NotificationService);
+    statusMessageService = TestBed.inject(StatusMessageService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should mark notification as read when button is clicked', () => {
+    jest.spyOn(notificationService, 'getAllNotificationsForTargetUser').mockReturnValue(of({
+      notifications: [testNotificationOne],
+    }));
+    jest.spyOn(notificationService, 'getReadNotifications').mockReturnValue(of({
+      readNotifications: [],
+    }));
+    const spy1: SpyInstance = jest.spyOn(notificationService, 'markNotificationAsRead')
+    .mockImplementation((request: MarkNotificationAsReadRequest) => {
+      expect(request.notificationId).toEqual(testNotificationOne.notificationId);
```

Comment:
indentation

Reply:
Thanks for the comment. I will fix all in the next commit (the same indentation will be updated for the admin side test).